lib/http/client.cpp

```diff
@@ -1660,6 +1660,18 @@ Client::set_pusher(const mtx::requests::SetPusher &req, Callback<mtx::responses:
       "/client/v3/pushers/set", req, std::move(cb));
 }
 
+void
+Client::search_user_directory(const std::string &search_term,
+                              Callback<mtx::responses::Users> callback,
+                              int limit)
+{
+    nlohmann::json req = {{"search_term", search_term}};
+    if (limit >= 0)
```

Comment:
I guess it is fine, but don't you think a value of 0 is funky? What's the intended use? Check if users are present without fetching any?

Reply:
The intended use of passing a limit of 0 is to get 1 result from synapse.

Reply:
Ah, that is terrible, but makes sense :D